packages/frontend/src/hooks/useCampaign.ts

```diff
@@ -0,0 +1,58 @@
+import type { SupportedChain } from "@metrom-xyz/contracts";
+import { type Campaign } from "@metrom-xyz/sdk";
+import { useEffect, useState } from "react";
+import type { Address } from "viem";
+import { metromApiClient, SUPPORTED_AMM_SLUG_TO_NAME } from "../commons";
+
+export interface NamedCampaign extends Campaign {
+    name: string;
+}
+
+export function useCampaign(
```

Comment:
Should we accept undefined params here and handle them inside the inner `useEffect`?

Reply:
Fixed in 328249fe1c2965f43e5177cfc2abbaf5b371501d.